packages/react-bindings/src/styles/getStyles.ts

```diff
@@ -74,12 +74,19 @@ const getStyles = (options: GetStylesOptions): GetStylesResult => {
     : undefined
 
   // Resolve styles using resolved variables, merge results, allow props.styles to override
-  const mergedStyles: ComponentSlotStylesPrepared = mergeComponentStyles(
-    theme.componentStyles[displayName],
-    props.design && withDebugId({ root: props.design }, 'props.design'),
-    props.styles && withDebugId({ root: props.styles } as ComponentSlotStylesInput, 'props.styles'),
-    animationStyles && withDebugId({ root: animationStyles }, 'props.animation'),
-  )
+  let mergedStyles: ComponentSlotStylesPrepared = theme.componentStyles[displayName] || {
```

Comment:
As the theme is immutable, the mergedStyles by default will always reference to the same object.